### PR TITLE
Change CRIU support from default off to default on

### PR DESCRIPTION
**Pull request by adrianreber**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

With the graduation of "Forensic Container Checkpointing" from Alpha to Beta the Kubernetes checkpoint feature defaults to on.

One of the arguments when introducing checkpoint/restore in CRI-O to default to off was to have the same behaviour as Kubernetes.

With this commit the behaviour of CRI-O is adapted to Kubernetes to default checkpoint/restore to on with the option to disable it explicitly.

CRI-O will also disable the feature if the CRIU binary is not found during initial CRI-O startup.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?
```release-note
CRI-O's checkpoint/restore support defaults to on mirroring the changes in Kubernetes.
```

### CODE REVIEW

**Comment by adrianreber** [comment]
I will fix the CI errors soon.

**Comment by adrianreber** [comment]
The CI error seems to be unrelated to my change.

**Review by saschagrunert — APPROVED** [review]
LGTM, are we happy with this @cri-o/cri-o-maintainers?

/hold
feel free to unhold if so

**Pull request merged**
